core/src/main/java/org/infinispan/protostream/impl/RawByteArrayOutputStreamImpl.java

```diff
@@ -0,0 +1,93 @@
+package org.infinispan.protostream.impl;
+
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+
+import net.jcip.annotations.NotThreadSafe;
+
+@NotThreadSafe
+public class RawByteArrayOutputStreamImpl extends OutputStream implements org.infinispan.protostream.RawByteArrayOutputStream {
```

Comment:
nitpicking
```suggestion
public class RawByteArrayOutputStreamImpl extends OutputStream implements RawByteArrayOutputStream {
```

Reply:
Why extend `OutputStream`?

Reply:
It's not strictly necessary, but it makes changing/benchmarking implementations easier when `OutputStream` is a common base class.